payment-request/algorithms-manual.https.html

```diff
@@ -0,0 +1,192 @@
+<!doctype html>
+<meta charset="utf8">
+<link rel="help" href="https://w3c.github.io/payment-request/#algorithms">
+<title>
+  Payment Request algorithms
+</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+setup({
+  explicit_done: true,
+  explicit_timeout: true,
+});
+const methods = [
+  {
+    supportedMethods: ["basic-card"],
+  },
+];
+const shippingOptions = {
+  shippingOptions: [
+    {
+      id: "fail",
+      label: "Option 1",
+      amount: {
+        currency: "USD",
+        value: "5.00",
+      },
+      selected: true,
+    },
+    {
+      id: "pass",
+      label: "Option 2",
+      amount: {
+        currency: "USD",
+        value: "5.00",
+      },
+    },
+  ],
+};
+
+const detailsNoShippingOptions = {
+  total: {
+    label: "Total due",
+    amount: {
+      currency: "USD",
+      value: "1.0",
+    },
+  },
+};
+
+const detailsWithShippingOptions = Object.assign(
+  {
+    total: {
+      label: "Total due",
+      amount: {
+        currency: "USD",
+        value: "1.0",
+      },
+    },
+  },
+  shippingOptions
+);
+
+const options = {
+  requestShipping: true,
+};
+
+test(() => {
+  const request = new PaymentRequest(
+    methods,
+    detailsNoShippingOptions,
+    options
+  );
+}, "Smoke test");
+
+function testFireEvent(button, details, eventName, expectRequestProps) {
+  button.disabled = true;
+  promise_test(async t => {
+    const request = new PaymentRequest(methods, details, options);
+    const events = [];
+    const p1 = new Promise(resolve => {
+      request[`on${eventName}`] = event => {
+        events.push(event);
+        resolve("handler");
+        event.updateWith(details);
+      };
+    });
+    const p2 = new Promise(resolve => {
+      request.addEventListener(eventName, event => {
+        events.push(event);
+        resolve("listener-1");
+      });
+    });
+    const p3 = new Promise(resolve => {
+      request.addEventListener(eventName, event => {
+        events.push(event);
+        resolve("listener-2");
+      });
+    });
+    try {
+      const response = await request.show();
+      response.complete("success");
+    } catch (err) {
+      await request.abort();
```

Comment:
I guess no need to call "await request.abort()", request.show() should execute about() internally once there is any error during showing. Wouldn't "await request.abort()" cause another errors? such as "InvalidStateError".

Reply:
yeah, good point. I'll remove.